src/Workspaces/Remote/Core/RemoteHostAssetSerialization.cs

```diff
@@ -97,5 +284,13 @@ internal static class RemoteHostAssetSerialization
                 callback(checksum, (T)result, arg);
             }
         }
+
+        static (byte, int) ReadSentinelAndLength(ReadResult readResult)
+        {
+            var sequenceReader = new SequenceReader<byte>(readResult.Buffer);
+            Contract.ThrowIfFalse(sequenceReader.TryRead(out var sentinel));
+            Contract.ThrowIfFalse(sequenceReader.TryReadLittleEndian(out int length));
```

Comment:
these cannot fail ('return false') as we call `ReadAtLeastAsync(HeaderSize,` right above this.  so we must be able to read out these 5 bytes.